### PR TITLE
Allow env variable to disable operator in Meshery instance

### PR DESCRIPTION
**Pull request by Revolyssup**
**Notes for Reviewers**

This PR fixes #
1.  Refactor code to avoid global variable
2. Add env variable to disable deployment of operator

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [ ] Yes, I signed my commits.

### CODE REVIEW

**Review by leecalcote — CHANGES_REQUESTED** [review]
Contributor and User documentation is missing.

**Comment by leecalcote** [comment]
@Azanul you may well want to add some thoughts here.

**Comment by Azanul** [comment]
The variable and method names seem confusing to me. Names like `OPERATOR_STATUS`, `SetOperatorStatus` seem to fit better. At least `Deploy`/`Undeploy` for function and `Deployed`/`Undeployed` for the booleans.

File: install/playground/meshery-playground-deployment.yaml [hunk 1]
```diff
@@ -24,6 +24,8 @@ spec:
               value: Meshery
             - name: PLAYGROUND
               value: "true"
+            - name: DISABLE_OPERATOR
```
Comment (Azanul) [inline install/playground/meshery-playground-deployment.yaml:27]: .

File: server/models/meshery_controllers.go [hunk 2]
```diff
@@ -171,12 +171,50 @@ func (mch *MesheryControllersHelper) UpdateOperatorsStatusMap(isUndeployed map[s
...
+	ctxIDtoDeploymentStatus map[string]bool
+	mx                      sync.Mutex
+	DisableOperator         bool
+}
+
+func NewOperatorTracker(disabled bool) *OperatorTracker {
+	return &OperatorTracker{
+		ctxIDtoDeploymentStatus: make(map[string]bool),
+		mx:                      sync.Mutex{},
+		DisableOperator:         disabled,
```
Comment (Azanul) [inline server/models/meshery_controllers.go:184]: .

File: server/models/meshery_controllers.go [hunk 3]
```diff
@@ -171,12 +171,50 @@ func (mch *MesheryControllersHelper) UpdateOperatorsStatusMap(isUndeployed map[s
...
+	}
+}
+
+func (ot *OperatorTracker) Undeployed(ctxID string, undeployed bool) {
```
Comment (Azanul) [inline server/models/meshery_controllers.go:188]: .

**Review by Azanul — COMMENTED** [review]
These are the ones I am talking about

**Comment by Azanul** [comment]
I recommend using `SetOperatorStatus` because I plan on using an enum for tracking status of connections. But we can do that with the state machine implementation.

File: server/models/meshery_controllers.go (hunk 3 above)
Comment (Azanul) [reply server/models/meshery_controllers.go:188]: `SetUndeployed` does feel like an improvement on `Undeployed`. How about `SetDeploymentStatus`?

**Comment by l5io** [comment]
🚀 Preview for commit 366d0e004e252f4ed3b3b7a2856dd0158489168d at: https://63ff495a84b8a9006cada6de--meshery-docs-preview.netlify.app

**Review by leecalcote — COMMENTED** [review]
Please open a separate issue to track the ability to disable the operator for a given meshconfig context. // @abdullah1308

**Review by leecalcote — APPROVED** [review]
Thanks for these docs.

**Comment by Revolyssup** [comment]
> Please open a separate issue to track the ability to disable the operator for a given meshconfig context. // @abdullah1308

https://github.com/meshery/meshery/issues/7233

**Comment by leecalcote** [comment]
Did you gents conclude?

**Pull request merged**